File: qr_login_manager.cs
// qr_login_manager.cs — C# версия

using System;
using System.IO;
using System.Text.Json;
using QRCoder;

class TokenData
{
    public string Token { get; set; }
    public string Timestamp { get; set; }
    public int Ttl { get; set; }
    public string Expires { get; set; }
}

class Program
{
    static void Main(string[] args)
    {
        int ttl = 60;
        string outputFile = "qrcode.png";
        string tokenFile = "token.json";
        if (args.Length > 0) ttl = int.Parse(args[0]);
        if (args.Length > 1) outputFile = args[1];
        if (args.Length > 2) tokenFile = args[2];

        Console.WriteLine("🔐 QR Login Manager (C#)");
        string token = Guid.NewGuid().ToString();
        string now = DateTime.UtcNow.ToString("o");
        string expires = DateTime.UtcNow.AddSeconds(ttl).ToString("o");

        Console.WriteLine($"✅ Токен сгенерирован: {token}");
        Console.WriteLine($"⏳ Срок действия: {ttl} секунд");
        Console.WriteLine("📱 Отсканируйте QR-код для входа:\n");

        var data = new TokenData { Token = token, Timestamp = now, Ttl = ttl, Expires = expires };
        string qrData = JsonSerializer.Serialize(data);

        // Генерация QR
        QRCodeGenerator qrGenerator = new QRCodeGenerator();
        QRCodeData qrCodeData = qrGenerator.CreateQrCode(qrData, QRCodeGenerator.ECCLevel.Q);
        QRCode qrCode = new QRCode(qrCodeData);
        var bitmap = qrCode.GetGraphic(20);
        bitmap.Save(outputFile, System.Drawing.Imaging.ImageFormat.Png);
        Console.WriteLine($"💾 Сохранено PNG: {outputFile}");

        // Сохранение токена
        File.WriteAllText(tokenFile, JsonSerializer.Serialize(data, new JsonSerializerOptions { WriteIndented = true }));
        Console.WriteLine($"💾 Сохранено токен: {tokenFile}");
    }
}

File: qr_login_manager.php
<?php
// qr_login_manager.php — PHP версия

require_once 'vendor/autoload.php'; // для endroid/qr-code

use Endroid\QrCode\Builder\Builder;
use Endroid\QrCode\Writer\PngWriter;
use Endroid\QrCode\Encoding\Encoding;
use Endroid\QrCode\ErrorCorrectionLevel\ErrorCorrectionLevelLow;

$ttl = isset($argv[1]) ? (int)$argv[1] : 60;
$outputFile = isset($argv[2]) ? $argv[2] : 'qrcode.png';
$tokenFile = isset($argv[3]) ? $argv[3] : 'token.json';

echo "🔐 QR Login Manager (PHP)\n";
$token = uniqid('', true);
$now = date('c');
$expires = date('c', time() + $ttl);

echo "✅ Токен сгенерирован: $token\n";
echo "⏳ Срок действия: $ttl секунд\n";
echo "📱 Отсканируйте QR-код для входа:\n\n";

$data = ['token' => $token, 'timestamp' => $now, 'ttl' => $ttl, 'expires' => $expires];
$qrData = json_encode($data);

// Генерация QR с помощью endroid/qr-code
$result = Builder::create()
    ->writer(new PngWriter())
    ->data($qrData)
    ->encoding(new Encoding('UTF-8'))
    ->errorCorrectionLevel(new ErrorCorrectionLevelLow())
    ->size(300)
    ->build();

// Сохранение PNG
$result->saveToFile($outputFile);
echo "💾 Сохранено PNG: $outputFile\n";

// Сохранение токена
file_put_contents($tokenFile, json_encode($data, JSON_PRETTY_PRINT));
echo "💾 Сохранено токен: $tokenFile\n";

// Вывод ASCII (простой вариант - только для демонстрации, обычно не выводим)
// Можно использовать библиотеку для ASCII, но для простоты пропустим.
?>
